pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -233,28 +288,46 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 					KubernetesServiceName:      svc.Attributes.Name,
 				},
 			}
-			tcpChain := &listener.FilterChain{
-				Filters: lb.buildInboundNetworkFilters(cc),
-				Name:    cc.clusterName,
-			}
+			var tcpChain, httpChain *listener.FilterChain
+			origDst := svc.GetAddressForProxy(lb.node) + ":" + portString
 			httpClusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP, "http", svc.Hostname, port.Port)
-			cc.clusterName = httpClusterName
-			httpChain := &listener.FilterChain{
-				Filters: lb.buildWaypointInboundHTTPFilters(svc, cc),
-				Name:    cc.clusterName,
+			if len(svcAddresses) > 0 {
```

Comment:
I don't really get this. Why does a service having VIPs mean we need getOrigDstSfs but otherwise we don't?

When does a service even have no VIPs and is valid to be including in ambient -- I think today a waypoint can never serve them?

Reply:
Are you saying to just always set the filter state regardless? Yeah I can buy that

Reply:
Maybe? I don't really understand at this point so not giving a recommendation yet :slightly_smiling_face:  I am not sure I understand what the VIP count has to do with hostname handling at all.

Reply:
I don't get it. A service without a VIP cannot be attached to a waypoint AFAIK.

Are you intentionally trying to changing that?

Reply:
No? What are you expecting to see? An early loop exit if there's no addresses for the service?

Reply:
Ah I think I see what you mean. So you actually want tighter controls so that services with no VIPs can't be served by the waypoint (at least not yet).